nsv13/code/modules/overmap/weapons/weapons.dm

```diff
@@ -64,18 +64,25 @@
 		return FIRE_MODE_TORPEDO
 	return FIRE_MODE_MAC
 
-/obj/structure/overmap/proc/select_weapon(number)
-	if(number > 0 && number <= length(weapon_numkeys_map))
-		swap_to(weapon_numkeys_map[number])
+/obj/structure/overmap/proc/select_weapon(number, wrole)
+	if(wrole && number > 0 && number <= length(weapon_numkeys_map))//GUNNER GO HERE, I SWAP FUNNI GUN
+		swap_to(weapon_numkeys_map[number], 1)
+		return TRUE
+	if(!wrole && number > 0 && number <= length(weapon_numkeys_map_pilot))//PILOT GO HERE, I SWAP FUNNI PILOT GUN
+		swap_to(weapon_numkeys_map_pilot[number], 2)
 		return TRUE
 
-/obj/structure/overmap/proc/swap_to(what=FIRE_MODE_ANTI_AIR)
+/obj/structure/overmap/proc/swap_to(what=FIRE_MODE_ANTI_AIR, num)
 	if(!weapon_types[what])
 		return FALSE
 	var/datum/ship_weapon/SW = weapon_types[what]
-	if(!(SW.allowed_roles & OVERMAP_USER_ROLE_GUNNER))
-		return FALSE
-	fire_mode = what
+	if(!(SW.allowed_roles & OVERMAP_USER_ROLE_GUNNER))//If your not a gunner and allowed role, continue
```

Comment:
Then this part would just become
`if(!(SW.allowed_roles & role))`
`    return FALSE`